applications/FluidDynamicsApplication/python_scripts/apply_two_fluids_inlet_process.py

```diff
@@ -0,0 +1,219 @@
+            self.aux_process_fluid_2 = assign_vector_by_direction_process.AssignVectorByDirectionProcess(Model, settings["fluid_2_settings"])
+
+
+
+    def ExecuteInitializeSolutionStep(self):
+        # Call the base process ExecuteInitializeSolutionStep()
+        if ( self.inlet_model_part.GetSubModelPart("fluid_1_inlet").NumberOfNodes() > 0):
+            self.aux_process_fluid_1.ExecuteInitializeSolutionStep()
+        if ( self.inlet_model_part.GetSubModelPart("fluid_2_inlet").NumberOfNodes() > 0):
+            self.aux_process_fluid_2.ExecuteInitializeSolutionStep()
+
+        # Not sure if appropriate here...
+        # PROBLEM: Could distort the physical properties
+        # BUT: It could make sense to stabilize cases with CFL >> 1
+        self.two_fluid_inlet_process.SmoothDistanceField()
+
+
+
+    def ExecuteFinalizeSolutionStep(self):
+        # Call the base process ExecuteFinalizeSolutionStep()
+        if ( self.inlet_model_part.GetSubModelPart("fluid_1_inlet").NumberOfNodes() > 0):
+            self.aux_process_fluid_1.ExecuteInitializeSolutionStep()
+        if ( self.inlet_model_part.GetSubModelPart("fluid_2_inlet").NumberOfNodes() > 0):
+            self.aux_process_fluid_2.ExecuteInitializeSolutionStep()
+
+        self.two_fluid_inlet_process.SmoothDistanceField()
+
+
+
+    def _set_variational_distance_process_serial(self):
+        # Construct the variational distance calculation process
+        maximum_iterations = 5
+
+        ### for serial and OpenMP
+        serial_settings = KratosMultiphysics.Parameters("""
+        {
+            "linear_solver_settings"   : {
+                "solver_type" : "AMGCL"
+            }
+        }
+        """)
+        import linear_solver_factory
+        self.linear_solver = linear_solver_factory.ConstructSolver(serial_settings["linear_solver_settings"])
+
+        if self.complete_model.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
+            variational_distance_process = KratosMultiphysics.VariationalDistanceCalculationProcess2D(
+                self.complete_model,
+                self.linear_solver,
+                maximum_iterations,
+                KratosMultiphysics.VariationalDistanceCalculationProcess2D.CALCULATE_EXACT_DISTANCES_TO_PLANE,
+                "distance_from_inlet_2D")
+        else:
+            variational_distance_process = KratosMultiphysics.VariationalDistanceCalculationProcess3D(
+                self.complete_model,
+                self.linear_solver,
+                maximum_iterations,
+                KratosMultiphysics.VariationalDistanceCalculationProcess3D.CALCULATE_EXACT_DISTANCES_TO_PLANE,
+                "distance_from_inlet_3D")
+
+        return variational_distance_process
+
+
+
+    def _set_variational_distance_process_mpi(self):
+        # Construct the variational distance calculation process
+        import KratosMultiphysics.TrilinosApplication as KratosTrilinos
```

Comment:
This kind of imports might be a source of problems. My suggestion is to create an `apply_two_fluids_inlet_process_mpi`. which of course must derive from this one, to reimplement all the MPI related stuff.

Reply:
Ok. Actually I think I can just override the function `set_variational_distance_process`. The rest is exactly the same, isn't it? It was done this way.